test/framework/robot.py

```diff
@@ -567,6 +567,14 @@ def test_get_toolchain_hierarchy(self):
             {'name': 'gompi', 'version': '1.4.10'},
```

Comment:
It's the caching for toolchain hierarchy screwing up the tests... We'll need to reset the cache.